core/src/main/java/io/ultrabrew/metrics/data/BasicCounterAggregator.java

```diff
@@ -16,43 +19,50 @@
  *
  * @see Counter
  */
-public class BasicCounterAggregator extends ConcurrentMonoidHashTable {
+public class BasicCounterAggregator extends ConcurrentMonoidLongTable {
 
   private static final String[] FIELDS = {"sum"};
   private static final Type[] TYPES = {Type.LONG};
   private static final long[] IDENTITY = {0L};
-  private static final int DEFAULT_CAPACITY = 128;
+
+  /**
+   * Create a monoid for common aggregation functions for a Counter.
+   * @param counter metric
+   */
+  public BasicCounterAggregator(final Counter counter) {
+    this(counter.id, counter.cardinality, counter.maxCardinality);
+  }
 
   /**
    * Create a monoid for common aggregation functions for a Counter.
    *
    * @param metricId identifier of the metric associated with this aggregator
    */
   public BasicCounterAggregator(final String metricId) {
-    this(metricId, DEFAULT_CAPACITY);
+    this(metricId, DEFAULT_CARDINALITY);
   }
 
   /**
    * Create a monoid for common aggregation functions for a Counter with requested capacity.
    *
    * @param metricId identifier of the metric associated with this aggregator
-   * @param capacity requested capacity of table in records, actual capacity may be higher
+   * @param cardinality requested capacity of table in records, actual capacity may be higher
    */
-  public BasicCounterAggregator(final String metricId, final int capacity) {
-    super(metricId, capacity, FIELDS, TYPES, IDENTITY);
+  public BasicCounterAggregator(final String metricId, final int cardinality) {
```

Comment:
I would still make this method `public BasicCounterAggregator(final String metricId, final int maxCardinality)` instead and calculate the initial capacity from the maxCardinality or use sane default.